config/config.go

```diff
@@ -664,6 +671,43 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string                    `yaml:"region"`
+	Access          string                    `yaml:"access,omitempty"`
+	Secret          string                    `yaml:"secret,omitempty"`
+	RefreshInterval Duration                  `yaml:"refresh_interval,omitempty"`
+	Port            int                       `yaml:"port"`
+	Tags            map[string]string         `yaml:"tags,omitempty"`
+	RegexTags       map[string]*regexp.Regexp `yaml:"-"`
+	Public          bool                      `yaml:"public,omitempty"`
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultEC2SDConfig
+	type plain EC2SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	if c.Port == 0 {
```

Comment:
I followed used the dns retrival as a template, which does exactly this. Happy to set the default to 80, though.